### PR TITLE
Disable aarch64 for some packages

### PR DESCRIPTION
**Pull request by annop-w**
This resolves issue #41148

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
LGTM

**Comment by tldahlgren** [comment]
Will defer merge to give @amd-toolchain-support time to comment.

**Review by amd-toolchain-support — APPROVED** [review]
Thanks, Tamara, for letting us know about this change.

We approve this change request.

**Comment by amd-toolchain-support** [comment]
@annop-w 
Instead of a plain "Not available on Aarch64" message
Request to update the conflict message section as - "AMD <respective library_name> available only on x86_64"

**Comment by annop-w** [comment]
> @annop-w Instead of a plain "Not available on Aarch64" message Request to update the conflict message section as - "AMD available only on x86_64"

Done.

**Review by amd-toolchain-support — APPROVED** [review]
Thanks.

File: var/spack/repos/builtin/packages/amdblis/package.py [hunk 1]
```diff
@@ -34,6 +34,8 @@ class Amdblis(BlisBase):
 
     maintainers("amd-toolchain-support")
 
+    conflicts("target=aarch64:", msg="AMD blis available only on x86_64")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/amdblis/package.py]: What about the other targets? Can't we:
```suggestion
    requires("target=x86_64:", msg="AMD blis available only on x86_64")
```
? Would that work?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/amdblis/package.py (hunk 1 above)
Comment (annop-w) [reply var/spack/repos/builtin/packages/amdblis/package.py]: Thank you for your suggestion. Done.

**Review by alalazo — APPROVED** [review]

**Pull request merged**